internal/gitstream/assign.go

```diff
@@ -108,14 +108,22 @@ func (a *Assign) assignIssues(ctx context.Context) error {
 
 			logger.Info("Assigning issue")
 
-			user, err := a.UserHelper.GetUser(ctx, upstreamCommit)
-			if err != nil {
-				return fmt.Errorf("could not get the upstream commit author for downstream issue %d: %v", *issue.Number, err)
-			}
-
-			assignee, err := owners.getAssignee(ctx, a.GC, *user.Login)
-			if err != nil {
-				return fmt.Errorf("could not get select a suitable assignee: %v", err)
+			var (
+				assignee string
+				intErr   error
+			)
+			if user, err := a.UserHelper.GetUser(ctx, upstreamCommit); err != nil {
+				logger.Info("WARNING: could not get the upstream commit author for downstream issue, picking a random assignee",
```

Comment:
How do we know if we failed to find the assignee because it does not exist on GitHub, or because there was e.g. a connectivity or API error?
We could change `GetUser` to return a well-known error and use `errors.Is()` to determine that.

Reply:
> That's OK IMO.

IMO, it's not OK to pick a random user because GitHub was down.

> on the next iteration we may to re-assign a new assignee for that specific commit.

How will we reassign if we skip issues that are already assigned?

Reply:
We won't. ATM we go over all the commits and do all the assignments or non of them (but not some yes and some no).

Reply:
So even with that change, if we assign an issue to a random user, we won't reassign later if we picked the wrong one, right?
I don't see the benefit of assigning a random user if there was an out of our control error while looking for the username.

Reply:
Fixed.